comit/Cargo.toml

```diff
@@ -48,5 +48,10 @@ testcontainers = "0.10"
 tokio = { version = "0.2", features = ["macros"] }
 
 [features]
-default = []
+default = ["test-docker"]
 test = ["libp2p/yamux", "libp2p/noise"]
+
+# "test-docker" feature is related to test code
+# if it's enabled then tests needing docker will be ran
+test-docker = []
```

Comment:
What do you think of just making `testcontainers` optional? Then you get a feature "for-free" by cargo and you don't have to declare one.

Unfortunately, dev-dependencies can't be optional so we'd have to move it to the normal dependencies. It would be good to make it non-default then, otherwise the dependency will be included in a normal build.

I know you had some concerns about wanting to run all tests locally with a single command. However, I would argue that idiomatic configuration is more important. If you care strongly about the "single command" feature, then feel free to add a Makefile entry or alternatively make yourself a local alias.

Another viewpoint onto this would be that running testcontainers-based tests is slow anyway, hence I _personally_ actually don't mind them being excluded by default in a `cargo test` run, similarly how running e2e tests is a dedicated command.

Reply:
I am sold.

It's not like I run much test locally anyway 🙃